Sources/SQLKit/Expressions/SQLStatement.swift

```diff
@@ -7,62 +53,190 @@ extension SQLSerializer {
+    /// The individual expressions collected by the statement, in order.
+    ///
+    /// The serialization of a given ``SQLStatement`` is that of each element of its ``parts`` array, with a
+    /// single space character placed between the SQL text of each element.
     public var parts: [any SQLExpression] = []
     
+    /// The ``SQLDatabase`` obtained from the original ``SQLSerializer``.
     @usableFromInline
     let database: any SQLDatabase
 
+    /// Designated initializer.
+    ///
+    /// External users may not invoke this method; use ``SQLSerializer/statement(_:)``.
     @usableFromInline
     init(database: any SQLDatabase) {
         self.database = database
     }
 
+    /// Convenience accessor for the database's `Logger`.
+    ///
+    /// > Note: The compiler's exclusive access checking rules prevent statement closures from accessing the
+    /// > original serializer directly.
+    @inlinable
+    public var logger: Logger {
+        self.database.logger
+    }
+
     /// Convenience accessor for the database's ``SQLDialect``.
     ///
-    /// - Note: Statement closures can't access the serializer directly due to exclusive access rules.
+    /// > Note: The compiler's exclusive access checking rules prevent statement closures from accessing the
+    /// > original serializer directly.
     @inlinable
     public var dialect: any SQLDialect {
         self.database.dialect
     }
 
-    /// Add raw text to the output.
+    // See `SQLExpression.serialize(to:)`.
+    @inlinable
+    public func serialize(to serializer: inout SQLSerializer) {
+        /// Although `self.parts.interspersed(with: SQLRaw(" ")).forEach { $0.serialize(to: &serializer) }` would be a
+        /// more "elegant" way to write this, it results in the creation of `self.parts.count - 1` identical instances
+        /// of ``SQLRaw`` and requires the compiler to dynamically dispatch a call to each one's `serialize(to:)`
+        /// method. While the total overhead of this behavior is unlikely to be measurable in practice unless the
+        /// statement has a very large number of constitutent parts, saving a couple of extra lines of code with a
+        /// "clever trick" is still not at all worth it - especially since it also requires importing the
+        /// `swift-algorithms` package, an entire additional dependency which adds insult to injury in the form of
+        /// increased overall compile time.
+        var iter = self.parts.makeIterator()
+        
+        iter.next()?.serialize(to: &serializer)
+        while let part = iter.next() {
+            var temp = SQLSerializer(database: serializer.database)
+            temp.binds = serializer.binds
+            
+            part.serialize(to: &temp)
+            if !temp.sql.isEmpty {
+                serializer.sql += " \(temp.sql)"
+            }
+            serializer.binds = temp.binds//.append(contentsOf: temp.binds) // Can't just append because we need to keep numbers in sync
```

Comment:
Do we want to remove the comment out code?

Reply:
I left it in for illustrative purposes, but I'm okay with removing it.